### ISSUE TITLE
[Bug]: Memory leak on v:lua.__bufferline_private.handle_click

### ISSUE THREAD

**Issue by bekirisgor** [akinsho/bufferline.nvim#632]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Memory leak on mouse click handler

packer installer 

use({ "akinsho/bufferline.nvim", tag = "v3.*", requires = "kyazdani42/nvim-web-devicons" })

### What did you expect to happen?

no memory leak :)

### Config

local status_ok, bufferline = pcall(require, "bufferline")
if not status_ok then
  return
  end
bufferline.setup {
  options = {
    modified_icon = "●",
    max_name_length = 30,
    max_prefix_length = 30, 
    tab_size = 21,
    diagnostics = false, 
    diagnostics_update_in_insert = false,
    offsets = { { filetype = "NvimTree", text = "", padding = 1 } },
    show_buffer_icons = true,
     show_buffer_close_icons = false,
    show_close_icon = false,
    show_tab_indicators = false,
    persist_buffer_sort = true, 
    separator_style = 'slant', 
    enforce_regular_tabs = false,
    always_show_bufferline = true,
  },
	highlights = require("catppuccin.groups.integrations.bufferline").get()
}

### Additional Information

![image](https://user-images.githubusercontent.com/17153168/206420307-63fbbb25-ddba-461b-8cae-259663d8bcff.png)

**Comment by bekirisgor** [comment]
Ok i reinstalled it without tag v3. The leak look likes gone.

**Comment by bekirisgor** [comment]
I supposed to its gone but still have memory leak. Currently, I disabled plugin

**Comment by akinsho** [comment]
@bekirisgor I'm currently busy with other non FOSS work please feel free to raise a PR and I can merge it if you are able to find and fix the cause

**Comment by bekirisgor** [comment]
> @bekirisgor I'm currently busy with other non FOSS work please feel free to raise a PR and I can merge it if you are able to find and fix the cause

I have tried to detect but i have dont have much experience about lua and neovim internal api. 

I temporarily disabled make_clickable function at - https://github.com/akinsho/bufferline.nvim/blob/main/lua/bufferline/ui.lua#L165

**Comment by akinsho** [comment]
@bekirisgor can you confirm if this is still happening and if so can you provide more details about your setup. How did you detect the leak? what version of neovim are you using? etc.

**Comment by bekirisgor** [comment]
> @bekirisgor can you confirm if this is still happening and if so can you provide more details about your setup. How did you detect the leak? what version of neovim are you using? etc.

Yes i still have memory leak but it doesnt expose the onclick function i mentioned at issue.  The leaks occurs when i click tabs.

I have been using forked one that disabled onclick functions. 

Feel free to close issue if it takes your time. I am fine without mouse interaction.

Here is my neovim version.  
```
NVIM v0.8.3
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by brew@Ventura-arm64.local
```

leaks cli tool output 

```Process:         nvim [31106]
Path:            /opt/homebrew/*/nvim
Load Address:    0x104784000
Identifier:      nvim
Version:         0
Code Type:       ARM64
Platform:        macOS
Parent Process:  fish [62580]

Date/Time:       2023-02-21 19:08:50.980 +0300
Launch Time:     2023-02-21 19:08:31.506 +0300
OS Version:      macOS 13.1 (22C65)
Report Version:  7
Analysis Tool:   /usr/bin/leaks

Physical footprint:         37.7M
Physical footprint (peak):  37.7M
Idle exit: untracked
----

leaks Report Version: 4.0
Process 31106: 20637 nodes malloced for 4894 KB
Process 31106: 12 leaks for 576 total leaked bytes.

    12 (576 bytes) << TOTAL >>
      1 (48 bytes) ROOT LEAK: 0x600001d47b40 [48]
      1 (48 bytes) ROOT LEAK: 0x600001d47c30 [48]
      1 (48 bytes) ROOT LEAK: 0x600001d700f0 [48]
      1 (48 bytes) ROOT LEAK: 0x600001d70c90 [48]
      1 (48 bytes) ROOT LEAK: 0x600001d70d50 [48]
      1 (48 bytes) ROOT LEAK: 0x600001d71a40 [48]
      1 (48 bytes) ROOT LEAK: 0x600001d71e60 [48]
      1 (48 bytes) ROOT LEAK: 0x600001d720d0 [48]
      1 (48 bytes) ROOT LEAK: 0x600001d72d60 [48]
      1 (48 bytes) ROOT LEAK: 0x600001d74360 [48]
      1 (48 bytes) ROOT LEAK: 0x600001d755f0 [48]
      1 (48 bytes) ROOT LEAK: 0x600001d7c6f0 [48]
```
